SK.WebApp/src/store/session/sessionSlice.ts

```diff
@@ -0,0 +1,47 @@
+import { createSelector, createSlice } from "@reduxjs/toolkit";
+import { RootState } from '..';
+
+interface SessionState {
+  username: string;
+  token: string;
+  image: string;
+};
+
+const initialState : SessionState = {
+  username: "",
+  token: "",
+  image: "",
+};
+
+export const sessionSlice = createSlice({
+    initialState,
+    name: 'session',
+    reducers: {
+      saveLoginData: (state, action) => {
+        if (action?.payload) {
+            state.username = action.payload.username;
+            state.token = action.payload.token;
+            state.image = action.payload.image;
+        }
+      }
+    },
+  });
+
+export const sessionSelector = createSelector<
+  RootState,
+  SessionState['username'],
+  SessionState['token'],
+  SessionState['image'],
+  SessionState
+>(
+  (state) => state.session.username,
+  (state) => state.session.token,
+  (state) => state.session.image,
+  (username, token, image) => {
+    return {
+      username,
+      token,
+      image,
+    };
+  },
+);
```

Comment:
I'm pretty sure this selector is not needed, this just adds complexity, nothing more 😄 If you need to select a session you'd do

```ts
const session = useSelector((state) => state.session);
```

Reply:
I removed this